arraycontext/impl/pytato/__init__.py

```diff
@@ -296,17 +296,7 @@ def _record_leaf_ary_in_dict(key: Tuple[Any, ...],
                     raise TypeError(f"{type(self).__name__}.freeze invoked "
                                     f"with non-pytato array of type '{type(array)}'")
 
-                if subary.size == 0:
```

Comment:
How big a win was the early exit?

Reply:
This wasn't put in for the sake of performance, but empty domains were interfering with loopy transforms, which is no longer an issue post https://github.com/inducer/pytato/pull/320.